data/transactions/logic/jsonspec_test.go

```diff
@@ -0,0 +1,336 @@
+	text = `{"key0": -Inf}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{"key0": null}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+}
+
+func TestParseHexValue(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	text := `{"key0": 0x1}`
+	_, err := parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{"key0": 0xFF}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+}
+
+func TestParseBigNum(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	//max int can be represented using float64, 2<<52
+	text := `{"key0":9007199254740992}`
+	msg, err := parseJSON([]byte(text))
+	require.Nil(t, err)
+	require.True(t, json.Valid([]byte(text)))
+	require.Equal(t, uint64(9007199254740992), uint64(msg.(map[string]interface{})["key0"].(float64)))
+
+	//2<<52+1
+	text = `{"key0":9007199254740993}`
+	msg, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	require.True(t, json.Valid([]byte(text)))
+	require.NotEqual(t, uint64(9007199254740993), uint64(msg.(map[string]interface{})["key0"].(float64)))
+	require.Equal(t, uint64(9007199254740992), uint64(msg.(map[string]interface{})["key0"].(float64)))
+}
+
+func TestParseExp(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	text := `{"key0": 1.2E+}`
+	_, err := parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{"key0": 1.2E+8}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": 0E+8}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": 0.2E+8}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": 0.2E-3}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": 1.2E-6}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+}
+
+func TestParseArrays(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	text := `{"key0": [,1,]}`
+	_, err := parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = "{\"key0\":[1\n]}"
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": [[1]]}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+}
+
+func TestParseKeys(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	text := `{"": 1}`
+	_, err := parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"": "algo"}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": 1,"key0": 2}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0": "1","key0": "2"}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"keys.1": 1}`
+	_, err = parseJSON([]byte(text))
+	require.Nil(t, err)
+	text = `{"key0":: 1}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{"key0":: "1"}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{"key0": 'algo'}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+	text = `{1: 1}`
+	_, err = parseJSON([]byte(text))
+	require.NotNil(t, err)
+
+}
+
+func TestParseFileEncoding(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	//utf-8
```

Comment:
Can you describe these test somewhat?  It seems like you might be addresses some of my concerns about encodings, but I don't know that I'm following everything.

Reply:
added a description.